source/slang/slang-ir.h

```diff
@@ -729,6 +729,16 @@ struct IRPtrLit : IRConstant
     void* getValue() { return value.ptrVal; }
 };
 
+// Range of keys of an RTTI entry.
+// For an `RTTIEntry(op0, op1)`, `op0` is an `IRIntLit` whose value
+// must be one of the enums defined in `RTTIEntryKeys`.
+enum RTTIEntryKeys : int32_t
+{
+    kRTTISize, // Size in bytes of the type represented by the RTTI object.
+    kRTTIKeys_End, // Marks the count of possible key values.
+                   // Not a valid value to appear in the IR.
```

Comment:
This is a rathole of naming convention stuff, but:

* My strong prefenece is that `enum` type names should be singular: e.g., `RTTIEntryKey` because when used as a type they represent a single value (this makes function signatures read better).

* My personally preference is that non-`enum class` enumerants should be of the form `k<TypeName>_<CaseName>`. So `kRTTIEntryKey_Size` would be my preference.

* There are dueling conventions in the codebase today about how to name the sentinel value past the end. I tend to use `k<TypeName>Count` while Jonathan tends to use `k<TypeName>_CountOf` (which doesn't parse grammatically for me...). This change would add a third option of `k<TypeName>_End` assuming you meant to write the full type name instead of just `RTTIKeys`...

Reply:
This is removed. We are using decorations instead of these entries so they are no longer needed.